akka-http-backend/src/test/scala/sttp/client/akkahttp/SttpBackendStubAkkaTests.scala

```diff
@@ -0,0 +1,83 @@
+package sttp.client.akkahttp
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.ws.{Message, TextMessage}
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.client._
+import sttp.model.Headers
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+class SttpBackendStubAkkaTests extends AnyFlatSpec with Matchers with ScalaFutures with BeforeAndAfterAll {
+
+  implicit val system: ActorSystem = ActorSystem()
+
+  override protected def afterAll(): Unit = {
+    Await.result(system.terminate().map(_ => ()), 5.seconds)
+  }
+
+  "backend stub" should "cycle through responses using a single sent request" in {
+    // given
+    implicit val backend = AkkaHttpBackend.stub
+      .whenRequestMatches(_ => true)
+      .thenRespondCyclic("a", "b", "c")
+
+    // when
+    def r = basicRequest.get(uri"http://example.org/a/b/c").send().futureValue
+
+    // then
+    r.body shouldBe Right("a")
+    r.body shouldBe Right("b")
+    r.body shouldBe Right("c")
+    r.body shouldBe Right("a")
+  }
+
+  it should "use given flow as web socket handler" in {
+
+    val echoFlow = Flow[Message]
+      .mapAsync(1) {
+        case tm: TextMessage => tm.toStrict(1.second)
+        case _               => Future.failed(new Exception("Expected TextMessage"))
+      }
+      .map { tms =>
+        TextMessage(s"echo: ${tms.text}")
+      }
+      .watchTermination()(Keep.right)
+
+    val useHandler: Flow[Message, Message, Future[Done]] => Future[Done] = clientFlow => {
+      clientFlow.via(echoFlow).via(clientFlow).runWith(Source(List(TextMessage("abc"))), Sink.ignore)._2
+    }
+
+    val clientFlow: Flow[Message, Message, Future[Seq[Message]]] = Flow.fromSinkAndSourceMat(
+      Sink.seq[Message],
+      Source((1 to 3).map(i => TextMessage(s"test$i")))
+    )(Keep.left)
+
+    implicit val b = AkkaHttpBackend.stub
+      .whenRequestMatches(_ => true)
+      .thenHandleOpenWebSocket(
+        Headers(List.empty),
+        useHandler
+      )
+
+    basicRequest
+      .get(uri"wss://echo.websocket.org")
+      .openWebsocket(clientFlow)
+      .flatMap(_.result)
+      .onComplete {
+        case Failure(exception) => println(s"WebSocket failure: $exception")
+        case Success(results) =>
+          println(results)
```

Comment:
can't we assert on the results that it contains the expected messages?

Reply:
I've reworked it but forgot to push after commit, sorry.